### PR TITLE
Feat/add scss namespaces

### LINKED ISSUE #52: Support Sass namespaced variables
Sass now has a [module system with `@use` and `@forward`](https://sass-lang.com/documentation/at-rules/use). With this new syntax, we can use variables and functions like:

```scss
@use '@primer/css/support/index' as primer;

.btn {
    color: primer.$text-gray-dark;
    background: primer.light-gradient()
```

Currently I get the following error: `Expected variable, function or keyword for "primer.$text-gray-dark" of "color"`.

---

Looking at the code, this is because the variables regex doesn’t account for that new syntax:

https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/197e257fb528015496b9d9ea4e062b7a28896053/src/index.js#L13-L15

Interestingly the functions regex already supports the new syntax 😄

Thinking of how to solve this – this is the regex I came up with, trying my best to add the new pattern without making it all slower by introducing overlaps between the patterns:

```regex
/^-?(?:@.+|\$.+|var\(--.+\)|(?!@|\$|var\(--).+\.\$.+)$/
```

https://regexper.com/#%2F%5E-%3F%28%3F%3A%40.%2B%7C%5C%24.%2B%7Cvar%5C%28--.%2B%5C%29%7C%28%3F!%40%7C%5C%24%7Cvar%5C%28--%29.%2B%5C.%5C%24.%2B%29%24%2F

regexper

If that sounds like a viable approach I would be happy to make a PR for this, testing the performance a bit further and writing tests for the new pattern.

### PR DESCRIPTION
**Pull request by AndyOGo**
fixes #52

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const messages = utils.ruleMessages(ruleName, {
   expected,
 })
 const reSkipProp = /^(?:@|\$|--).+$/
-const reVar = /^-?(?:@.+|\$.+|var\(--.+\))$/
+const reVar = /^-?(?:@.+|(?:[^(]+\.)?\$.+|var\(--.+\))$/
```
Comment (thibaudcolas) [inline src/index.js:14]: This pattern works but it has the potential to introduce false positives due to accepting "any characters". For example:

```css
.test::before {
  content: ".$5";
}
```

I’m not sure how much it matters considering how unlikely it is for a CSS value to contain `.$` 🤔.

From what I could find [in the spec](https://github.com/sass/sass/blob/master/accepted/module-system.md#member-references) it looks like the namespace is meant to be a valid CSS [ident-token](https://drafts.csswg.org/css-syntax-3/#ident-token-diagram), so a more bulletproof regex might be:

```regex
/^-?(?:@.+|(?:[a-zA-Z0-9-_]+\.)?\$.+|var\(--.+\))$/
```

This is more permissive than the actual definition of ident-token from the spec, but that shouldn’t really be a problem for the purpose of this plugin.

Comment (AndyOGo) [reply src/index.js:14]: Thank you for your instant feedback 👍 
You are right, all patterns in this plugin are loose and very permissive.
So far a duck on purpose^^

`content` rule is a very good point 👍 
I should extend all tests with that rule.

Anyway as you said, it leads to false positives.
Therefor a proper CSS properties/values parser would be the best solution.

I looked already into `postcss-values-parser`, which supports atm only standard CSS Syntax, so Sass and Less is still missing.
Though changing that would be a major release.

File: src/index.js [hunk 2]
```diff
@@ -11,7 +11,11 @@ const messages = utils.ruleMessages(ruleName, {
   expected,
 })
 const reSkipProp = /^(?:@|\$|--).+$/
-const reVar = /^-?(?:@.+|\$.+|var\(--.+\))$/
+// Sass namespaces and CSS <ident-token>
+// @see: https://github.com/sass/sass/blob/master/accepted/module-system.md#member-references
+// @see:  https://drafts.csswg.org/css-syntax-3/#ident-token-diagram
+// eslint-disable-next-line no-control-regex
+const reVar = /^-?(?:@.+|(?:(?:[a-zA-Z_-]|[^\x00-\x7F])+(?:[a-zA-Z0-9_-]|[^\x00-\x7F])*\.)?\$.+|var\(--.+\))$/
```
Comment (AndyOGo) [inline src/index.js:18]: @thibaudcolas 
Alright, I I added an `<ident-token>` regex for the sass namespace, includeing Non-ASCII range.
Thank you!

Comment (AndyOGo) [inline src/index.js:18]: @thibaudcolas
Alright, I I added an <ident-token> regex for the sass namespace, includeing Non-ASCII range.
Thank you!

Comment (thibaudcolas) [reply src/index.js:18]: Nice!

**Pull request merged**